### ISSUE TITLE
Allow forced empty commits and merges

### ISSUE THREAD

**Issue by arielshaqed** [treeverse/lakeFS#7042]
Allow empty commits by setting some optional "allow-empty" property.  This should be possible both on both commits and merges.  **Keep backwards compatibility and be like Git:** By default continue to _forbid it_, only allow it if explicitly requested.

## Uses

* Any periodic commit operation: the empty commit records the fact that the process occurred and committed, and that it added no new data.
* Update mtime or other undigested fields, to create a new commit.
* Simplify automated tasks: it is often easier to commit than to commit and then handle one particular error.
* (Merges): Record that a merge was tried but had nothing to do.

## 🦕 Ancient history 🦖 

Note that this is partially the reverse of #1172.

<details>
 (I like to think of that set of fixes before we released Graveler as a group effort, please don't look to see who opened it)
</details>

  At the time we wanted to keep the Git behaviour of failing an empty commit by default.  IIRC we didn't want to add an "allow empty commits" flag, for fear that it would never be used.

Well, now we have a user :-)

## Context: Slack discussion

This [user request here](https://lakefs.slack.com/archives/C016726JLJW/p1700736838542909?thread_ts=1700640298.594399&cid=C016726JLJW) is for something potentially larger, but by allowing just this we can get the desired behaviour.  I believe that this proves it's great tooling!

**Comment by arielshaqed** [comment]
We've also received occasional other queries about this very feature; sorry, I have no way of finding them.

**Comment by itaiad200** [comment]
@treeverse/product

**Comment by eladlachmi** [comment]
Kilian Brandt (Toyota) asked about this feature today:

> when we create branch there is a exist_ok flag.   
> It could be nice to have a empty_ok flag for commit that would just return a reference on  
> HEAD if there is nothing to commit.

>We have a common usecase where some data change, our pipeline re-run, not everything will lead to updates and therefore some commits might be empty

They migrated to the new Python SDK this week, so the new flag will need to be reflected there.

**Comment by johnnyaug** [comment]
There are two senses in which a _merge_ can be empty:

1. The source and destination branch **point to the same commit**. In such cases we show an error, and I think it's fine - even though Git allows it with an `Already up to date.` message.
2. The source has only empty commits after departing from the destination - after we allow empty commits, this will be a valid repo state. In this case, I think we will always allow the merge, and I think that's also fine.

To sum up, I think no need to add anything to the merge flow.
